src/pages/[platform]/build-a-backend/data/custom-business-logic/search-and-aggregate-queries/index.mdx

```diff
@@ -0,0 +1,369 @@
+
+Important considerations:
+
+Ensure Point in Time Recovery (PITR) is enabled, which is crucial for the pipeline integration.
+Enable DynamoDB streams to capture item changes that will be ingested into OpenSearch.
+
+```ts title="amplify/backend.ts"
+
+const todoTable =
+  backend.data.resources.cfnResources.amplifyDynamoDbTables["Todo"];
+
+todoTable.pointInTimeRecoveryEnabled = true;
+
+todoTable.streamSpecification = {
+  streamViewType: dynamodb.StreamViewType.NEW_IMAGE,
+};
+
+```
+
+## Step 1: Setting Up the OpenSearch Instance
+
+Create an OpenSearch instance with encryption.
+
+``` ts title="amplify/backend.ts"
+const openSearchDomain = new opensearch.Domain(
+  openSearchStack,
+  "OpenSearchDomain",
+  {
+    version: opensearch.EngineVersion.OPENSEARCH_2_11,
+    nodeToNodeEncryption: true,
+    encryptionAtRest: {
+      enabled: true,
+    },
+  }
+);
+
+```
+
+## Step 2: Setting Up Zero ETL from DynamoDB to OpenSearch
+
+### Step 2a: S3 Bucket and IAM Role
+Create an S3 bucket for backing up raw events consumed by the OpenSearch pipeline. Also, set up an IAM role for the pipeline:
+
+``` ts title="amplify/backend.ts"
+
+const s3BackupBucket = new s3.Bucket(
+  openSearchStack,
+  "OpenSearchBackupBucketAmplifyGen2",
+  {
+    blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+    bucketName: "opensearch-backup-bucket-amplify-gen-2-test",
+    enforceSSL: true,
+    versioned: true,
+    autoDeleteObjects: true,
+    removalPolicy: RemovalPolicy.DESTROY,
+  }
+);
+
+const openSearchIntegrationPipelineRole = new iam.Role(
+  openSearchStack,
+  "OpenSearchIntegrationPipelineRole",
+  {
+    assumedBy: new iam.ServicePrincipal("osis-pipelines.amazonaws.com"),
+    inlinePolicies: {
+      openSearchPipelinePolicy: new iam.PolicyDocument({
+        statements: [
+          new iam.PolicyStatement({
+            actions: ["es:DescribeDomain"],
+            resources: [openSearchDomain.domainArn],
+            effect: iam.Effect.ALLOW,
+          }),
+          new iam.PolicyStatement({
+            actions: ["es:ESHttp*"],
+            resources: [openSearchDomain.domainArn],
+            effect: iam.Effect.ALLOW,
+          }),
+        ],
+      }),
+    },
+    managedPolicies: [
+      iam.ManagedPolicy.fromAwsManagedPolicyName(
+        "AmazonOpenSearchServiceFullAccess"
+      ),
+    ],
+  }
+);
+
+openSearchIntegrationPipelineRole.addManagedPolicy(
+  iam.ManagedPolicy.fromAwsManagedPolicyName("AmazonDynamoDBFullAccess")
+);
+openSearchIntegrationPipelineRole.addManagedPolicy(
+  iam.ManagedPolicy.fromAwsManagedPolicyName("AmazonS3FullAccess")
+);
+
+```
+For the S3 bucket, follow standard security practices: block public access, encrypt data at rest, and enable versioning.
+
+The IAM role should allow the OpenSearch Ingestion Service (OSIS) pipelines to assume it. Grant specific OpenSearch Service permissions and also provide DynamoDB and S3 access. You may customize permissions to follow the principle of least privilege.
```

Comment:
"You may customize permissions to follow the principle of least privilege"

let's have the sample show this